Framework/src/MonitorObjectCollection.cxx

```diff
@@ -120,6 +121,41 @@ const std::string& MonitorObjectCollection::getTaskName() const
   return mTaskName;
 }
 
+std::string formatDuration(uint64_t durationMs)
+{
+  auto remainder = durationMs;
+  uint64_t hours = remainder / (1000 * 60 * 60);
+  remainder %= (1000 * 60 * 60);
+  uint64_t minutes = remainder / (1000 * 60);
+  remainder %= (1000 * 60);
+  uint64_t seconds = remainder / 1000;
+
+  std::stringstream result;
+
+  if (hours > 0) {
+    result << hours << "h";
+  }
+  if (minutes > 0 || hours > 0) {
+    result << minutes << "m";
+  }
+  result << seconds << "s";
+  if (durationMs < 1000) {
+    result << durationMs << "ms";
+  }
```

Comment:
the ms are ignored if the duration is more than 1 second, correct ?

Reply:
Correct. I don't think there is need for such verbosity if the validity is longer. It's just for shifters to be aware what they are looking at.

Reply:
I agree